neonix/io/nixio.py

```diff
@@ -0,0 +1,270 @@
+        parent_block after converting to a ``Group`` object.
+
+        :param segment: Neo segment to be written
+        :param parent_block: The parent NIX Block
+        :return: The newly created NIX Group
+        """
+        nix_name = segment.name
+        nix_type = "neo.segment"
+        nix_definition = segment.description
+        nix_group = parent_block.create_group(nix_name, nix_type)
+        nix_group.definition = nix_definition
+        if segment.rec_datetime:
+            # Truncating timestamp to seconds
+            nix_group.force_created_at(int(segment.rec_datetime.timestamp()))
+        if segment.file_datetime:
+            group_metadata = self._get_or_init_metadata(nix_group)
+            # Truncating timestamp to seconds
+            group_metadata .create_property(
+                    "neo.file_datetime",
+                    nix.Value(int(segment.file_datetime.timestamp())))
+        if segment.file_origin:
+            group_metadata = self._get_or_init_metadata(nix_group)
+            group_metadata.create_property("neo.file_origin",
+                                           nix.Value(segment.file_origin))
+        return nix_group
+
+    def add_recordingchannelgroup(self, rcg, parent_block):
+        """
+        Write the provided ``rcg`` (RecordingChannelGroup) to the NIX file as
+        a child of ``parent_block`` after converting to a ``Source`` object.
+
+        :param rcg: The Neo rcg to be written
+        :param parent_block: The parent neo block of the provided rcg
+        :return: The newly created NIX Source.
+        """
+        nix_name = rcg.name
+        nix_type = "neo.recordingchannelgroup"
+        nix_definition = rcg.description
+        nix_source = parent_block.create_source(nix_name, nix_type)
+        nix_source.definition = nix_definition
+        if rcg.file_origin:
+            source_metadata = self._get_or_init_metadata(nix_source)
+            source_metadata.create_property("neo.file_origin",
+                                            nix.Value(rcg.file_origin))
+        if hasattr(rcg, "coordinates"):
+            source_metadata = self._get_or_init_metadata(nix_source)
+            coordinates = rcg.coordinates  # should this be copied?
+            source_metadata.create_property("neo.coordinates",
+                                            nix.Value(coordinates))
+        return nix_source
+
+    def _get_or_init_metadata(self, nix_obj):
+        """
+        Creates a metadata Section for the provided NIX object if it doesn't
+        have one already. Returns the new or existing metadata section.
+
+        :param nix_obj: The object to which the Section is attached.
+        :return: The metadata section of the provided object.
+        """
+        if nix_obj.metadata is None:
+            nix_obj.metadata = self.nix_file.create_section(
+                    nix_obj.name, nix_obj.type+".metadata")
```

Comment:
Maybe it is nicer to build up a metadata / section tree that corresponds to the neo structure: the section holding metadata of a segment is a child section of the one that has the metadata for the parent block.